src/Generating/PrefabPiecePool.cpp

```diff
@@ -5,6 +5,81 @@
 
 #include "Globals.h"
 #include "PrefabPiecePool.h"
+#include "../Bindings/LuaState.h"
+#include "SelfTests.h"
+#include "WorldStorage/SchematicFileSerializer.h"
+
+
+
+
+
+// Conditionally log a warning
+#define CONDWARNING(ShouldLog, ...) \
+	if (ShouldLog) \
+	{ \
+		LOGWARNING(__VA_ARGS__); \
+	}
+
+
+
+
+
+#if SELF_TEST
+static class cPrefabPiecePoolTest
+{
+public:
+	cPrefabPiecePoolTest(void)
+	{
+		cSelfTests::Get().Register(cSelfTests::SelfTestFunction(cPrefabPiecePoolTest::TestLoading), "PrefabPiecePool loading test");
+	}
+
+	static void TestLoading(void)
+	{
+		cPrefabPiecePool test;
+		auto res = test.LoadFromFile("test.cubeset", true);
+		if (!res)
+		{
+			LOGWARNING("Loading from file \"test.cubeset\" failed.");
+			return;
+		}
+		LOG("Loaded %u pieces and %u starting pieces", static_cast<unsigned>(test.m_AllPieces.size()), static_cast<unsigned>(test.m_StartingPieces.size()));
+	}
+} g_Test;
```

Comment:
I think we should merge this and then play around with the tests. I'd like to continue the work on the actual loading, so that the generators use the loaded prefabs instead of hardcoded ones. If this is alright with you, then review the code as is and let's merge.